dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -0,0 +1,316 @@
+package com.dotmarketing.cms.urlmap;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.business.IdentifierAPI;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.cache.FieldsCache;
+import com.dotmarketing.cms.urlmap.filters.URLMapFilter;
+import com.dotmarketing.common.model.ContentletSearch;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.structure.StructureUtil;
+import com.dotmarketing.portlets.structure.factories.StructureFactory;
```

Comment:
We need to remove all references to legacy FieldsCache and Structure and StructureFactory